Shared/ViewModels/LibraryFilterViewModel.swift

```diff
@@ -0,0 +1,61 @@
+//
+/*
+ * SwiftFin is subject to the terms of the Mozilla Public
+ * License, v2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright 2021 Aiden Vigue & Jellyfin Contributors
+ */
+
+import Combine
+import Foundation
+import JellyfinAPI
+
+enum FilterType {
+    case tag
+    case genre
+    case sortOrder
+    case sortBy
+    case filter
+}
+
+final class LibraryFilterViewModel: ViewModel {
+    @Published
+    var modifyedFilters = LibraryFilters()
+
+    @Published
+    var allGenres = [NameGuidPair]()
+    @Published
+    var allTags = [String]()
+    @Published
+    var allSortOrders = APISortOrder.allCases
+    @Published
+    var allSortBys = SortBy.allCases
+    @Published
+    var allItemFilters = ItemFilter.allCases
```

Comment:
would prefer `possibleItemFilters` instead of `allItemFilters`
same for others in this area